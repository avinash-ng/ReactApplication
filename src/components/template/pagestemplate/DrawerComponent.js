import React, {useState} from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Drawer from '@material-ui/core/Drawer';
import CssBaseline from '@material-ui/core/CssBaseline';
import AppBar from '@material-ui/core/AppBar';
import ToolBar from '../../molecules/Toolbar/ToolBar';
import List from '@material-ui/core/List';
// import Typography from '@material-ui/core/Typography';
import Divider from '@material-ui/core/Divider';
import ListItem from '@material-ui/core/ListItem';
// import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
// import InboxIcon from '@material-ui/icons/MoveToInbox';
// import MailIcon from '@material-ui/icons/Mail';
// import ReactDOM from 'react-dom';
import { useAuth0 } from "../../../react-auth0-spa"
import Button from "@material-ui/core/Button"
import history from "../../../utils/history"
// import Title from "../../atoms/Title/Title"
import sinon from 'sinon'
import './DrawerComponent.css'
import MainListItems from '../../molecules/List/MainListItems';
import ListWithSubMenu from '../../organisms/ListWithSubMenu/ListWithSubMenu';
const drawerWidth = 240;




const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    height: 'inherit'
  },
  appBar: {
    width: `calc(100% - ${drawerWidth}px)`,
    marginLeft: drawerWidth,
  },
  drawer: {
    width: drawerWidth,
    flexShrink: 0,
  },
  drawerPaper: {
    width: drawerWidth,
  },
  // necessary for content to be below app bar
  toolbar: theme.mixins.toolbar,
  content: {
    marginTop: '64px',
    flexGrow: 1,
    height: 'inerit',
    backgroundColor: theme.palette.background.default,
    padding: theme.spacing(3),
  },
}));

const clicked = (text) => {
  var loweredText = "/" + text.toLowerCase();

  history.push(loweredText);
  // ReactDOM.render(, document.getElementById("content"));
}





export default function PermanentDrawerLeft({ children }) {
  
  const classes = useStyles();
  //hardcode user name
  
  // var userName = user===undefined? 'avinash': user.name
  const { user, logout, isAuthenticated } = useAuth0();
  const [anchorEl, setAnchorEl] = useState(null)

  const handleClick = event => {
    setAnchorEl(event.currentTarget);
  }
  const hoverHandler = () => {
    document.getElementsByClassName('list-div')[0].style.display = ''
  }
  const leaveHandler = () => {
      document.getElementsByClassName('list-div')[0].style.display = 'none'
  }


    return (
       
      <div className={classes.root} >
      <CssBaseline />
      <AppBar position="fixed" className={classes.appBar}>
          <ToolBar style={{ marginLeft: '1100px', width: '75px', backgroundColor: 'white' }} name={user.name} buttonName="logout" variant="contained" onClick={logout}/>
      </AppBar>
      <Drawer
        className={classes.drawer}
        variant="permanent"
        classes={{
          paper: classes.drawerPaper,
        }}
        anchor="left"
      >
        <div className={classes.toolbar} />
        <Divider />
          <List>
           
          {/* {['Home', 'About', 'Contact', 'Gallery'].map((text, index) => (
            <ListItem button style={{marginLeft: '0px'}} key={text} onClick={() => clicked(text)}>
              <ListItemText primary={text} />
            </ListItem>
          ))} */}
            
              <MainListItems clicked={clicked}/>
              {/* <div className="dropdown">
              <ListItem button key='Image' className="dropbtn">
                <ListItemText primary={'Image'} />
                </ListItem>
                <div className="dropdown-content">
                  <a href="/edit">Edit</a>
                  <a href="/delete">Delete</a>
                  <a href="/upload">Upload</a>
                  </div>
                </div>
              */}
            <ListWithSubMenu />
              </List> 
          

        <Divider />
          
        {/* <div className="list-div">
            <ul>
              <li>
                <Button> Edit </Button>
              </li>
              <li>
                <Button> Delete </Button>
              </li>
              <li>
                <Button>  </Button>
              </li>
            </ul>    
        </div> */}
          
          
        
      </Drawer>
      <main className={classes.content} id="main">
        {/* <div className={classes.toolbar} /> */}
        
          <div id="content"> {children}</div>
      </main>
    </div>
  );
}
