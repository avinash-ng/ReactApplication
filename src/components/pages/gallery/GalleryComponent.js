import React, { useEffect, useState } from 'react';
import DrawerComponent from "../../template/pagestemplate/DrawerComponent"
import axios from 'axios'
// import "./GalleryComponent.css"
// import Image from "../../atoms/Image/Image"
// import HTMLButton from "../../atoms/Button/Button"
// import ImageEditable from "../../molecules/image/ImageEditable"
import ImageOrganism from '../../organisms/image/ImageOrganism';
import {makeStyles} from '@material-ui/core/styles'

const useStyles = makeStyles((theme) => ({
    imageDiv: {
        width: '500px',
        height: '500px',
        float: 'left'
    }
}))

export const ContentComponent = (props) => {
    console.log("making request...")
    const classes = useStyles();
    const [images, updateImages] = useState([])
    const [value, updateValue] = useState('')
    const [clickedHeading, updateClickedHeading] = useState('')
    const [objectToBeChanged, updateObjectToBeChanged] = useState('')


    useEffect(() => {
        console.log("Inside useEffect...")
        axios({
            method: "GET",
            url: "http://localhost:3001/image",
            params: {
                id:2,
            },
            responseType: "json",
        }).then(response => {
            console.log(response)
            response.data.map(eachObject => ( eachObject.selected=false, updateImages(images => [...images, eachObject])))
        })
        props.updateState(true)
    }, [])

    const changeHeading = (obj, index) => {
        console.log(obj)
        console.log("is clicked...", index)
        var imagesCopy = [...images];
        imagesCopy[index].selected = true;
        updateImages(imagesCopy)
        // // updateClickedHeading(obj.name)
        console.log(obj)
        updateObjectToBeChanged({ ...objectToBeChanged, ...obj })
        // console.log("Inside change heading ", objectToBeChanged)
        // console.log(obj.name)
    }

    const confirmEdit = (each, index) => {

        var newFileName = document.getElementsByClassName("h3-image-tag")[index].innerHTML.split('.png')[0]
        var oldFileName = objectToBeChanged.name.split('.png')[0]
        console.log("New file name is ", newFileName, " old file name is ", oldFileName)
        console.log("Confirmed changes....")
        axios.patch("http://localhost:3001/image", {
            oldFileName: oldFileName,
            newFileName: newFileName,
        })
        var imagesCopy = [...images];
        imagesCopy[index].selected = false;
        updateImages(imagesCopy)
        updateObjectToBeChanged(null)
        window.location.reload();

    }
    
    return (
        <div>
            <ul style={{listStyleType: 'none'}}>
                {images.map((each, index) => (
                    <li key={index}>
                        {console.log("selected is " ,each.selected)}
                {/* <div className="img-div">
                            <Image altText="Loading error...." imageName={"http://localhost:3001/uploads/" + each.name}></Image>
                            <br />
                            <ImageEditable className="h3-image-tag" headingOnClicked={() => changeHeading(each, index)} object={each} buttonName="Rename" buttonOnClicked={()=> confirmEdit(each, index)}></ImageEditable>           
                </div> */}
                        <ImageOrganism divClassName="img-div" className={classes.imageDiv} imgClassName="h3-image-tag" imgAltText="Loading error..."
                            headingOnClicked={() => changeHeading(each, index)} object={each} buttonName="Save" buttonOnClicked={() => confirmEdit(each, index)}
                        ></ImageOrganism>
                </li>
            ))}
            </ul>
        </div>
    )

}
const GalleryComponent = () => {
    const [value, updateValue] = useState(false)
    const updateState = () => {
        updateValue(!value)
    }
    return (

        <div>
            <DrawerComponent> <ContentComponent updateState={updateState}/> </DrawerComponent>

        </div>
    )
}
export default GalleryComponent;





// value = { value } onChange = {(e) => onChangeHandler(each)}
{/* <div className="tag-div" contentEditable="true" suppressContentEditableWarning={true}> */ }
{/* <h3 className = "h3-image-tag" style={{ width: '450px' }}  onClick={() => changeHeading(each, index)} >{each.name}</h3> */ }
{/* <ImageTag className="h3-image-tag" onClicked={() => changeHeading(each, index)} object={each}></ImageTag> */ }
{/* {each.selected ? <HTMLButton buttonName="Rename" onClicked={() => confirmEdit(each, index)}></HTMLButton> : ""}  */ }

{/* </div> */ }