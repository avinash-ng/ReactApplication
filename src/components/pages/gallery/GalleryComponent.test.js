const { shallow, configure, mount } = require("enzyme")
import GalleryComponent from "./GalleryComponent"
import React from 'react'
import Adapter from 'enzyme-adapter-react-16';
import { useAuth0 } from '../../../react-auth0-spa'
import DrawerComponent from "../../template/pagestemplate/DrawerComponent";
import {ContentComponent} from './GalleryComponent'
configure({ adapter: new Adapter() });
const user = {
    email: "av@gmail.com",
    email_verified: true,
    sub: "google-oauth2|12345678901234"
}
jest.mock("../../../react-auth0-spa");
describe('testing <GalleryComponent /> ', () => {
    beforeEach(() => {
        // Mock the Auth0 hook and make it return a logged in state
        useAuth0.mockReturnValue({
            isAuthenticated: true,
            user,
            logout: jest.fn(),
            loginWithRedirect: jest.fn()
        });
    });

    it('renders <DrawerComponent />  correctly', () => {

        const RenderComponent = mount(<GalleryComponent> </GalleryComponent>)
        expect(RenderComponent.find(DrawerComponent).exists()).toBe(true)
    })

    it('renders <ContentConponent />  correctly', () => {

        const RenderComponent = mount(<GalleryComponent> </GalleryComponent>)
        expect(RenderComponent.find(ContentComponent).exists()).toBe(true)
    })

    it('matches snapshot', () => {
        const RenderComponent = shallow(<GalleryComponent ></GalleryComponent>)
        expect(RenderComponent).toMatchSnapshot()
    })
})