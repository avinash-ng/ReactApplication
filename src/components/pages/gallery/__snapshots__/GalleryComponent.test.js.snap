// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <GalleryComponent />  matches snapshot 1`] = `ShallowWrapper {}`;
